pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -557,3 +566,33 @@ func removeMaxAttachLimit(driverName string) nodeUpdateFunc {
 		return node, true, nil
 	}
 }
+
+// validateCSINodeInfo ensures members of CSINodeInfo object satisfies map and set semantics.
+// Before calling CSINodeInfoInterface.Create() or CSINodeInfoInterface.Update()
+// validateCSINodeInfo() should be invoked to make sure the CSINodeInfo is compliant
+// TODO: move this logic to an external webhook
+func validateCSINodeInfo(nodeInfo *csiv1alpha1.CSINodeInfo) error {
+	if len(nodeInfo.CSIDrivers) < 1 {
+		return fmt.Errorf("at least one CSI Driver entry is required")
+	}
+	// check for duplicate entries for the same driver
+	var errors []string
+	driverNames := make(sets.String)
+	for _, driverInfo := range nodeInfo.CSIDrivers {
```

Comment:
Since [this PR](https://github.com/kubernetes/kubernetes/pull/70515) merged, does this check need to be split to verify both Spec and Status?